packages/typespec-vscode/src/typespec-utils.ts

```diff
@@ -0,0 +1,60 @@
+import { readdir } from "node:fs/promises";
+import path from "path";
+import vscode from "vscode";
+import { StartFileName } from "./const.js";
+import logger from "./log/logger.js";
+import { getDirectoryPath, normalizeSlashes } from "./path-utils.js";
+import { isFile } from "./utils.js";
+
+export const onStdioOut = (str: string) => {
+  str
+    .trim()
+    .split("\n")
+    .forEach((line) => logger.info(line));
+};
+export const onStdioError = (str: string) => {
+  str
+    .trim()
+    .split("\n")
+    .forEach((line) =>
+      logger.error(line, [], {
+        showOutput: true,
+        showPopup: false,
+      }),
+    );
+};
+
+export async function getEntrypointTspFile(tspPath: string): Promise<string | undefined> {
```

Comment:
i think you need to search parent folder for the main.tsp too. you can refer to getMainFileForDocument in core\server\compiler-service.ts

Reply:
I don't think we need to search parent folder. This function is to retrieve out all the tsp project (main.tsp) under a specified folder.

Reply:
but what if i start from a tsp file in sub folder? i.e. in following structure, i right click on sub.tsp
- src
   main.tsp
   - subfolder
      sub.tsp <-

Reply:
If we support search parent folder, thing may become complex and un-control:

1. When the main.tsp in a parent folder which outside of the vscode workspace?
2. When there are more than one main.tsp in parent folder
   . src
      main.tsp
          subfolder1
             main.tsp
             subfolder2
                    sub.tsp

Reply:
we already have that logic, I think we can just follow the same. Please refer to getMainFileForDocument in core\server\compiler-service.ts

Reply:
For #2 scenario, we will select the closest main.tsp as entrypoint, consistent with logic in compiler.